workflow-job

```diff
@@ -261,50 +281,84 @@ flows["${sha1}"] = {
+
+              echo "Running integration tests on Cassandra on ${buildId}"
+              sh "${findImapPortCassandra} && docker run ${verbose} --name=${containers.integrationCassandra} --entrypoint=\"/root/integration_tests.sh\" ${useMavenCache} --volumes-from=${containers.gitPublish} ${images.jamesCompile} ${integrationArguments}"
+          }
         }
-        stage "Configuring James Server"
-        node {
-            sh "docker exec ${containers.james} ${jamesCliWithOptions} adddomain domain"
-            sh "docker exec ${containers.james} ${jamesCliWithOptions} adduser imapuser@domain password"
-             
-            echo "Running integration tests on ${buildId}"
-            sh "${findImapPort} && docker run ${verbose} --name=${containers.integration} --entrypoint=\"/root/integration_tests.sh\" ${useMavenCache} --volumes-from=${containers.gitPublish} ${images.jamesCompile} ${integrationArguments}"
-            testing.success();
+
+        deployAndValidatingBuilder["JPA"] = {
+          stage "Deploying Guice Server with JPA ${buildId}"
+          node {
+              sh "cd dockerfiles/run/guice/jpa; docker cp ${containers.keystore}:${keystorePath} destination/conf"
+              sh "cd dockerfiles/run/guice/jpa; docker cp ${containers.jamesCompile}:${jarPathJPA} destination"
+              sh "cd dockerfiles/run/guice/jpa; docker cp ${containers.jamesCompile}:${libPathJPA} destination"
+              sh "cd dockerfiles/run/guice/jpa; docker cp ${containers.jamesCompile}:${cliJarPathJPA} destination"
+              sh "cd dockerfiles/run/guice/jpa; docker cp ${containers.jamesCompile}:${clilibPathJPA} destination"
+              
+              sh "cd dockerfiles/run/guice/jpa; docker build --tag=${images.jamesJPA} ."
+              dockering.success()
+              sh "docker run --detach=true --name=${containers.jamesJPA} --hostname ${hostname} --expose=143 --publish-all=true ${images.jamesJPA}"
+
+              echo 'Waiting for James server to be deployed'
+              try {
+                  waitForCommandSuccess("docker exec ${containers.jamesJPA} ${jamesCliWithOptions} listusers", maxRetries)
+              } catch (Exception e) {
+                  sh "docker logs ${containers.jamesJPA}"
+              }
+          }
+          stage "Configuring James Server on JPA"
+          node {
+              provisionDataForIntegrationTest("${containers.jamesJPA}", "${jamesCliWithOptions}")
+
+              echo "Running integration tests on JPA on ${buildId}"
+              sh "${findImapPortJPA} && docker run ${verbose} --name=${containers.integrationJPA} --entrypoint=\"/root/integration_tests.sh\" ${useMavenCache} --volumes-from=${containers.gitPublish} ${images.jamesCompile} ${integrationArguments}"
+          }
         }
-        stage "Java 6 integration tests"
+
+        parallel deployAndValidatingBuilder
+        deploying.success();
```

Comment:
Nope. You should do these validations inside your parrallel branches. And we had a bug before : 
 - Marking deployement  as successed if Spring **OR** Guice succeed. It should be **AND**. But I belive we should rather add more details...

Please status definition should be : 

```
    def statuses = new CommitStatuses();
    def merging = new CommitStatus(repoName, sha1, "Merging", "1");
    statuses.addPendingStatus(merging);
    def building6 = new CommitStatus(repoName, sha1, "Building java 6", "2");
    statuses.addPendingStatus(building);
    def building8 = new CommitStatus(repoName, sha1, "Building java 8", "3");
    statuses.addPendingStatus(building);
    def dockeringGuiceCassandra = new CommitStatus(repoName, sha1, "Dockering Guice cassandra", "4");
    statuses.addPendingStatus(dockering);
    def dockeringGuiceJPA = new CommitStatus(repoName, sha1, "Dockering Guice JPA", "5");
    statuses.addPendingStatus(dockering);
    def dockeringSpringJPA = new CommitStatus(repoName, sha1, "Dockering Spring JPA", "6");
    statuses.addPendingStatus(dockering);
    def deployingGuiceCassandra = new CommitStatus(repoName, sha1, "Deploying Guice Cassandra", "7");
    statuses.addPendingStatus(deploying);
    def deployingGuiceJPA = new CommitStatus(repoName, sha1, "Deploying Guice JPA", "8");
    statuses.addPendingStatus(deploying);
    def deployingSpringJPA = new CommitStatus(repoName, sha1, "Deploying Spring JPA", "9");
    statuses.addPendingStatus(deploying);
    def testingGuiceCassandra = new CommitStatus(repoName, sha1, "Running integration tests on Guice Cassandra", "10");
statuses.addPendingStatus(testing);
    def testingGuiceJPA = new CommitStatus(repoName, sha1, "Running integration tests on Guice JPA", "11");
statuses.addPendingStatus(testing);
    def testingSpringJPA = new CommitStatus(repoName, sha1, "Running integration tests on Spring JPA", "12");
statuses.addPendingStatus(testing);
``` 

And you should mark them as success INSIDE branches.